### PR TITLE
Introducing: The Mega Horn

### PR DESCRIPTION
**Pull request by CRITAWAKETS**
Your new gateway to warcrimes!

## About The Pull Request

This PR adds in an admin-only clown superweapon in the form of the Mega Horn, a gigantic bike horn. In practice, its a handheld version of the Honkerblast that throws stuff around.

Also only clowns can use this thing.

## Why It's Good For The Game

Its not, but i love it anyway.

## Changelog
:cl:
add: Reports have been made of the Honkmother granting her funniest and most trusted disciples a bike horn that could end the world.
/:cl:

I wanted to make it stronger to the point of calling the shuttle, but admins are babies. _you can still change the cooldown time if you love fun though_

### CODE REVIEW

**Comment by itseasytosee** [comment]
admin only but role restricted, :shipit:

**Comment by MrDoomBringer** [comment]
haters will call it pointless bloat

**Comment by Tharcoonvagh** [comment]
Make it cost 25 tc and or *a lot of* bananium

**Comment by OverlyGeneric** [comment]
BFG but for clowns.

**Comment by Doosteroni** [comment]
You can't just honk a hole into the center of mars

**Comment by trollbreeder** [comment]
> You can't just honk a hole into the center of mars

Objective:
Honk a hole into the center of mars

File: code/game/objects/items/clown_items.dm [hunk 1]
```diff
@@ -235,3 +235,57 @@
...
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "mega_horn"
+	inhand_icon_state = "bike_horn"
+	worn_icon_state = "horn"
+	hitsound = "sound/items/bikehorn.ogg"
+	attack_verb_continuous = list("HONKS")
+	attack_verb_simple = list("HONK")
+	w_class = WEIGHT_CLASS_HUGE //This dude's a chonker.
+	COOLDOWN_DECLARE(mega_horn_cooldown)
+	var/cooldown_length = 80 //malicious admin laughing engage
```
Comment (Rohesie) [inline code/game/objects/items/clown_items.dm:251]:
```suggestion
	var/cooldown_length = 8 SECONDS //malicious admin laughing engage
```
* Use time defines.

File: code/game/objects/items/clown_items.dm [hunk 2]
```diff
@@ -235,3 +235,57 @@
...
+
+/obj/item/megahorn/attack_self(mob/source, atom/target, params) //this is just the honkerblast adapted for handheld use
+	var/mob/living/carbon/human/H = source
+	if (HAS_TRAIT(H, TRAIT_CLUMSY)) //only clowns have the power
```
Comment (Rohesie) [inline code/game/objects/items/clown_items.dm:255]: * Early return.

File: code/game/objects/items/clown_items.dm [hunk 3]
```diff
@@ -235,3 +235,57 @@
...
+		if(!COOLDOWN_FINISHED(src, mega_horn_cooldown))
+			return
+		playsound(src, 'sound/items/airhorn.ogg', 100, TRUE)
+		to_chat(source, "[icon2html(src, source)]<font color='red' size='5'>HONK</font>")
+		for(var/mob/living/carbon/M in ohearers(6, source))
```
Comment (Rohesie) [inline code/game/objects/items/clown_items.dm:260]: Will people on the same tile as the clown not be affected by it? Why?

File: code/game/objects/items/clown_items.dm [hunk 4]
```diff
@@ -235,3 +235,57 @@
...
+			var/obj/item/organ/ears/ears = M.getorganslot(ORGAN_SLOT_EARS)
+			if(ears)
+				ears.adjustEarDamage(0, 30)
+			M.Paralyze(60)
+			if(prob(30))
+				M.Stun(200)
+				M.Unconscious(80)
+			else
+				M.Jitter(500)
+		COOLDOWN_START(src, mega_horn_cooldown, cooldown_length) //A generous amount of time to think about what you have done.
```
Comment (Rohesie) [inline code/game/objects/items/clown_items.dm:282]: Start the cooldown right after the check, always as early as possible.

File: code/game/objects/items/clown_items.dm [hunk 5]
```diff
@@ -235,3 +235,57 @@
...
+
+		log_message("Honked from [src.name]. HONK!", LOG_ATTACK)
```
Comment (Rohesie) [inline code/game/objects/items/clown_items.dm:284]:
```suggestion
		log_message("Honked from [name]. HONK!", LOG_ATTACK)
```
`src` is implied, no need to use it.

File: code/game/objects/items/clown_items.dm [hunk 6]
```diff
@@ -235,3 +235,57 @@
...
+		var/turf/T = get_turf(src)
```
Comment (Rohesie) [inline code/game/objects/items/clown_items.dm:285]: Use better variable names than `T`

**Comment by Rohesie** [comment]
Garner the interest of at least an admin or two, have them comment on this PR supporting it, so we avoid unused item bloat.
If the admin team thinks this would have utility for their shenanigans it can be merged.

**Comment by Tharcoonvagh** [comment]
Oh well,

**Pull request closed without merging**